Work towards Gentoo build

thesamesam opened the pull request:
@radarhere I've got to the point where a build takes ~10m (including running Pillow's test suite). Needs more work to run Pillow from the mount i.e. /Pillow where the repo-state-to-be-tested-is but that part shouldn't be too bad.

Hopefully this is useful. I'm working on getting the s390x stages up to date so those can possibly be used via Docker too.

thesamesam committed:
gentoo: get base building & testing package

Not yet testing the current repository state though.

Signed-off-by: Sam James <sam@gentoo.org>

thesamesam committed:
gentoo: run tests on correct repository state

Signed-off-by: Sam James <sam@gentoo.org>

thesamesam commented:
It is now quite close but I don't understand the failure to import at the end.

radarhere commented:
Hi. As you may have realised, I didn't expect this to happen from my `gentoo` branch.
To tidy things up, I've put together [`test_gentoo`](https://github.com/radarhere/docker-images/tree/test_gentoo) instead
- I've removed my earlier commits, but kept yours
- I've changed the `pytest` command to `python3 -m pytest`, and [it works](https://github.com/radarhere/docker-images/runs/5271503618). I don't know why `pytest` didn't work by itself - my guess is maybe there's more than one pytest instance?
- I did remove your update to the Pillow repository itself. Now that I'm no longer removing everything else, Pillow will be updated to the latest version of that repository by https://github.com/radarhere/docker-images/blob/c36180611b8e8d77b47663736ac54ce495b1e4fe/.github/workflows/build.yml#L33-L34

thesamesam commented:
> Hi. As you may have realised, I didn't expect this to happen from my `gentoo` branch. To tidy things up, I've put together [`test_gentoo`](https://github.com/radarhere/docker-images/tree/test_gentoo?rgh-link-date=2022-02-21T09%3A17%3A25Z) instead

Apologies :)

Thank you for both merging and the tidying up!

> 
>     * I've removed my earlier commits, but kept yours
> 
>     * I've changed the `pytest` command to `python3 -m pytest`, and [it works](https://github.com/radarhere/docker-images/runs/5271503618). I don't know why `pytest` didn't work by itself - my guess is maybe there's more than one pytest instance?

Ah, it must be!

> 
>     * I did remove your update to the Pillow repository itself. Now that I'm no longer removing everything else, Pillow will be updated to the latest version of that repository by https://github.com/radarhere/docker-images/blob/c36180611b8e8d77b47663736ac54ce495b1e4fe/.github/workflows/build.yml#L33-L34

Oh, I see, that's easier.

thesamesam commented:
Please tell me if you need anything further from me in either making things work or keeping them working (this is a recurring offer, doesn't matter if it's in a while or not).

Gentoo being part of project CIs is valuable to us if it's doable. Even if it isn't run all the time, being available and set up is useful. Big thanks again.

radarhere commented:
This has now become https://github.com/python-pillow/docker-images/pull/141
